From 7ec71966765a7a9da81647e21b7d4767af8bba64 Mon Sep 17 00:00:00 2001
From: Ding Yunliang <yunliang.ding@intel.com>
Date: Fri, 1 Jun 2018 17:05:01 +0800
Subject: [PATCH 1967/2367] media: intel-ipu4: isys: make attribute packed work

Message for Open Source:
Fix clang build warning(like below):
ipu-isys-csi2.h:142:1: warning: attribute 'packed' is ignored, place
it after "struct" to apply attribute to type declaration
[-Wignored-attributes]

Message for Internal:
[Issue/Feature] Same as above.
[Root Cause/Changes] Same as above.

Change-Id: I074c911b183f4e39e8726c6f73a2b82b41bde267
Tracked-On: #JICCCN-228
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu-isys-csi2.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-isys-csi2.h b/drivers/media/pci/intel/ipu-isys-csi2.h
index 34c202f..10836c3 100644
--- a/drivers/media/pci/intel/ipu-isys-csi2.h
+++ b/drivers/media/pci/intel/ipu-isys-csi2.h
@@ -125,16 +125,16 @@ struct ipu_isys_csi2_timing {
  * from IPU MIPI receiver. Due to hardware conversion,
  * this structure is not the same as defined in CSI-2 spec.
  */
-__packed struct ipu_isys_mipi_packet_header {
+struct ipu_isys_mipi_packet_header {
 	u32 word_count:16, dtype:13, sync:2, stype:1;
 	u32 sid:4, port_id:4, reserved:23, odd_even:1;
-};
+} __packed;
 
 /*
  * This structure defines the trace message content
  * for CSI2 receiver monitor messages.
  */
-__packed struct ipu_isys_csi2_monitor_message {
+struct ipu_isys_csi2_monitor_message {
 	u64 fe:1,
 	    fs:1,
 	    pe:1,
@@ -153,7 +153,7 @@ __packed struct ipu_isys_csi2_monitor_message {
 	    port:4, vc:2, reserved4:2, frame_sync:4, reserved5:4;
 	u64 reserved6:3,
 	    cmd:2, reserved7:1, monitor_id:7, reserved8:1, timestamp_h:50;
-};
+} __packed;
 
 #define to_ipu_isys_csi2(sd) container_of(to_ipu_isys_subdev(sd), \
 					struct ipu_isys_csi2, asd)
-- 
2.7.4

